### PR TITLE
Deprecate 'cordova plugin search' command

### PR DESCRIPTION
**Pull request by erisu**
### Platforms affected
none

### What does this PR do?
- Remove `cordova plugin search` command
- Remove any reference of search command messaging
- Remove search command tests

### What testing has been done on this change?
- [x] `npm test`

### CODE REVIEW

**Comment by erisu** [comment]
@raphinesse Yes, to deprecate or deprecated does means it would still exist, still work with the current version, possibly replaced with something newer and more preferred, and will be removed in the future. 

I might have gone a bit too far and just happened to have removed it when I noticed it only opens the URL and with the plugin append to the query string if passed. Kinda feels like an anti-cli pattern where it navigates a user away from CLI.  `npm search` on the other hand displays results within CLI.

Anyways, If you and others do prefer to deprecate and display a warning message that it will be removed in a future release, I can revert all the changes and add a single `warning emit`. 

That would mean we can also close out https://github.com/apache/cordova-cli/pull/306 as won't merge.

No problems either way.

**Comment by raphinesse** [comment]
@erisu I just wanted to be sure we're all on the same page regarding terminology.

As I said before, since this is purely a convenience feature and I see no use in it either, I'm fine with removing this. And @dpogue seems to approve the removal as well.

Last time I tried to use the CLI search of npm, I ended up reading an issue where they said that it's been kind of broken forever :grin:

File: src/cordova/plugin/add.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ module.exports.listUnmetRequirements = listUnmetRequirements;
 
 function add (projectRoot, hooksRunner, opts) {
     if (!opts.plugins || !opts.plugins.length) {
-        return Q.reject(new CordovaError('No plugin specified. Please specify a plugin to add. See `' + cordova_util.binname + ' plugin search`.'));
+        return Q.reject(new CordovaError('No plugin specified. Please specify a plugin to add.'));
```
[the remark this answers is not in the document]
Comment (erisu) [reply src/cordova/plugin/add.js]: @raphinesse I am OK with a link, but truthfully though CLI, I know that `if` condition is actually never reached. I was debating on deleting it but would rather have a refactor cleanup PR.

Though CLI, if a plugin is not provided, THIS is the actual `if` condition that is caught and throws a different reject message.

https://github.com/apache/cordova-lib/blob/ea181ad553d1d2c93a968f1455109b776b92f0fa/src/cordova/plugin/index.js#L62

We could update this condition instead, except it is also a generic message for both `add` and `rm`. It also does not support `remove`. Just more things being uncovered as I been looking at this originally.

[the remark this answers is not in the document]
Comment (raphinesse) [reply src/cordova/plugin/add.js]: Ah, I see. Given these options, I think I'll got with the refactor cleanup PR :smile: 

My plan was to do a complete refactor of the CLI using `yargs` or something like that. Had no time to tackle that yet, though.

**Pull request merged**